mlx/fast.cpp

```diff
@@ -1,8 +1,14 @@
 // Copyright © 2023-2024 Apple Inc.
 
+#include <algorithm>
 #include <cassert>
+#include <iostream>
```

Comment:
Forgotten? I tend to do that a lot :-)

Reply:
Nah it is fine. I am not sure we would leave print in there. I would rather have the generated kernel stored or something if verbose, not sure... It is definitely good for now.

Also I just wrote the comment before looking at the print.

If I had to guess of a principled way of printing the kernel then I would make `verbose` to be of type `std::optional<std::ostream>` and if it has a value I would print there. This way one can save it in a string or print it from python and so on and so forth.

Reply:
> If I had to guess of a principled way of printing the kernel then I would make verbose to be of type std::optional<std::ostream> and if it has a value I would print there. This way one can save it in a string or print it from python and so on and so forth.

I like that suggestion more than logging to `std::cout`. Another option that might work is to expose a "kernel_string" attribute /method which gives you the generated kernel in a `std::string`.

Reply:
What do you want to do here @barronalex ? If we intend to change the API I would either leave the argument out for now (to make it easier to add) or change it.